build-aux/git-version-gen

```diff
@@ -152,7 +152,8 @@ then
 # derive a version string.
 elif test "`git log -1 --pretty=format:x . 2>/dev/null`" = x \
     && v=`git describe --tags --abbrev=7 --match="$prefix*" HEAD 2>/dev/null \
-          || git describe --tags --abbrev=7 HEAD 2>/dev/null` \
+          || git describe --tags --abbrev=7 HEAD 2>/dev/null \
+          || git log -1 --pretty=format:'v0-HEAD-%h' 2>/dev/null` \
```

Comment:
This won't work on some kinds of checkouts. I'll think about this one a bit...

Reply:
`git describe` has an `always` flag that returns a bare SHA if there is nothing else available:

```bash
$ git describe
fatal: No names found, cannot describe anything.
$ git describe --always
12d7610
```

Reply:
Also you may be able do a `git fetch --depth 100` or some number that is likely to be more than the time since the last release to try to get the tag.

Reply:
Doing a fetch in the build system is problematic because some environments don't have access to the credentials or network access for it. Also I'm not sure `--always` is what we want here because this is _supposed_ to fail if we aren't sure what we are.

Reply:
Okay I've done a bit of poking and `--always` does break what this is supposed to do in some scenarios, and ding a `git fetch` from inside our build system is out of the question, but this change as presented doesn't do any _harm_. It's slightly more information about a broken build environment than we would have otherwise. I still think the real solution is to make sure git based builds are never shallow clones, but this is all we can do from inside the SILE source.